agent-control/src/health/on_host/http.rs

```diff
@@ -105,30 +115,38 @@ impl HttpHealthChecker<InnerClient> {
 
 impl<C: HttpClient> HealthChecker for HttpHealthChecker<C> {
     fn check_health(&self) -> Result<HealthWithStartTime, HealthCheckerError> {
-        let response = self
-            .client
-            .get(self.url.as_str(), &self.headers)
-            .map_err(|e| HealthCheckerError::Generic(e.to_string()))?;
-        let status_code = response.status();
-
-        let status = String::from_utf8_lossy(response.body()).into();
-
-        if (self.healthy_status_codes.is_empty() && status_code.is_success())
-            || self.healthy_status_codes.contains(&status_code.as_u16())
-        {
-            return Ok(HealthWithStartTime::from_healthy(
-                Healthy::new().with_status(status),
-                self.start_time,
-            ));
+        match self.client.get(self.url.as_str(), &self.headers) {
+            Ok(response) => {
```

Comment:
previous code was not assuming a response had status code 2xx, is it the case now?